### DISCUSSION TITLE
Pumped Hydro Storage data from FERC 1

### DISCUSSION THREAD

**Discussion by AlvaroFurlani** [catalyst-cooperative/pudl#1752] [Help Me!] [answered]
Two questions related to energy storage systems data from FERC 1 forms:

1. Many of the entries in the f1_pumped_storage table are missing. For example, out of the 6167 entries, 'tot_capacity' is 0 for 5451 entries. Is this issue caused during the data integration into PUDL or does the FERC database is also missing those values?

2. What table contain data from the "Energy Storage Operations" section from FERC 1 forms?

**Comment by zaneselvans** [comment, marked as answer]
Whatever you're finding in the [f1_pumped_storage table](https://data.catalyst.coop/ferc1/f1_pumped_storage) should be exactly what was in the original FERC Form 1 databases that FERC published. We don't clean up or transform that data at all (except insofar as it's all put into a single SQLite DB instead of 27 separate Visual FoxPro DBs). The raw data often has a very large number of invalid records -- FERC made no effort to follow normal database design patterns.  It's just a container for what was previously a paper form.

The [plants_pumped_hydro_ferc1 table](https://data.catalyst.coop/pudl/plants_pumped_storage_ferc1) in the PUDL DB is our cleaned up version of the original data from FERC's DB. It only contains 710 records, but they should all be real data. It's possible that some data has been lost in that translation if it was split across multiple rows as sometimes is the case.

We're in the process of revamping some of the FERC 1 transform functions as we integrate FERC's new XBRL format (see #1665) so if you see some valuable data that's been lost in going from the messy original FERC 1 table to the cleaned up PUDL table, please let us know! 

Can you provide a more explicit pointer on the "Energy Storage Operations" section? What pages of the form is that related to?  We have [a directory mapping FERC 1 DB tables to pages](https://catalystcoop-pudl.readthedocs.io/en/latest/data_dictionaries/ferc1_db.html) in our documentation if that helps.

**Reply by AlvaroFurlani** [reply]
Thank you for the clarification. Regarding the second question, the "Energy Storage Operations" section is on page 414 of the FERC 1 form; I don't see it on the mapping table that you mentioned.

**Reply by zaneselvans** [reply]
What version of the Form 1 are you looking at? The PDF I've got (from [our archive here](https://catalystcoop-pudl.readthedocs.io/en/latest/data_sources/ferc1.html) which seems to match what I'm finding on the FERC website and which is supposedly valid through 2022-11-30) jumps from p411 to p422. Is this a new portion of the form meant to deal with battery storage too?

They haven't released the 2021 XBRL data in full yet, and we're not done building a system to parse and integrate it.
